src/index.js

```diff
@@ -1,45 +1,127 @@
-/* eslint-disable import/no-commonjs */
-const runes = require('runes');
+import runes from 'runes';
 
-const assertIsValidText = function(text) {
+const assertIsValidText = text => {
   if (typeof text !== 'string') {
     throw new TypeError(
       'Text should be provided as first argument and be a string.'
     );
   }
 };
 
-const assertIsValidChunkSize = function(chunkSize) {
-  if (typeof chunkSize !== 'number' || chunkSize <= 0) {
+const assertIsValidChunkSize = chunkSize => {
+  if (Number.isNaN(chunkSize) || Number.parseInt(chunkSize, 10) <= 0) {
     throw new TypeError(
-      'Size should be provided as 2nd argument and be a number greater than zero.'
+      'Size should be provided as 2nd argument and parseInt to a value greater than zero.'
     );
   }
 };
 
-export default function(text, chunkSize) {
-  assertIsValidText(text);
-  assertIsValidChunkSize(chunkSize);
-
-  const chunks = [];
-  let characters = runes(text);
+const assertIsValidChunkType = (
+  chunkType,
+  chunkTypeParseIntNaN,
+  chunkTypeInt
+) => {
+  if (
+    typeof chunkType !== 'undefined' &&
+    chunkType !== null &&
+    chunkType !== '' &&
+    (chunkTypeParseIntNaN || chunkTypeInt < -1)
+  ) {
+    throw new TypeError(
+      'Type should be provided as 3rd (optional) argument and parseInt to a value >= -1.'
+    );
+  }
+};
 
-  while (characters.length > chunkSize) {
-    const splitAt = characters.lastIndexOf(' ', chunkSize);
+const chunkLength = (characters, chunkType) => {
```

Comment:
there's a lot of syntactic changes, which makes it harder to review the change itself, is it possible to split that up in separate PRs, one for the syntax changes, and one for the functional change?

Reply:
Probably yeah! Haven't tried but should be easy to switch back. I saw your other suggestion to switch some back to regular functions, that make sense and can do, is there any specific time to prefer arrow function vs regular function? Should I port all of them back to regular functions?